dll/win32/shell32/CDefView.cpp

```diff
@@ -1215,11 +1265,12 @@ BOOL CDefView::_Sort(int Col)
     /* Set the sorting style on the new column */
     hColumn.mask = HDI_FORMAT;
     Header_GetItem(hHeader, m_sortInfo.ListColumn, &hColumn);
-    hColumn.fmt &= (m_sortInfo.bIsAscending ? ~HDF_SORTDOWN : ~HDF_SORTUP );
-    hColumn.fmt |= (m_sortInfo.bIsAscending ?  HDF_SORTUP   : HDF_SORTDOWN);
+    hColumn.fmt &= ~(HDF_SORTDOWN | HDF_SORTUP);
+    hColumn.fmt |= (m_sortInfo.Direction > 0 ?  HDF_SORTUP : HDF_SORTDOWN);
     Header_SetItem(hHeader, m_sortInfo.ListColumn, &hColumn);
 
     /* Sort the list, using the current values of ListColumn and bIsAscending */
+    ASSERT(m_sortInfo.Direction == 1 || m_sortInfo.Direction == -1);
```

Comment:
Why can't the direction be == 0 here?
(See line 1250 for when it is set to 0.)

Reply:
And on line 1253 it makes sure it is not 0 before doing the actual sort.